users/views.py

```diff
@@ -126,13 +127,20 @@ def get_object(self, queryset=None):
 
 class UserDetail(DetailView):
     slug_field = 'username'
+    template_name = 'users/user_detail.html'
 
     def get_queryset(self):
         queryset = User.objects.select_related()
         if self.request.user.username == self.kwargs['slug']:
             return queryset
         return queryset.searchable()
 
+    def get_object(self, queryset=None):
+        try:
+            return super().get_object(queryset)
+        except Http404:
+            return AnonymousUser()
```

Comment:
Would returning an anonymous user make it known that a user doesn't exist and the malicious actor could continue enumerating over usernames?

Reply:
Initially, yes - since the HTML contents used to include the user's name in the `<title>` tag - that's been removed in the template.

I examined the HTML output of an existing vs anonymous user to confirm that the only difference is in the actual URL requested - nothing else.

Reply:
> I examined the HTML output of an existing vs anonymous user to confirm that the only difference is in the actual URL requested - nothing else.

non-blocking suggestion: If it's not too expensive in terms of effort, it would be good to add a test that does this.